src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java

```diff
@@ -216,6 +222,35 @@ void shouldGetPreComputePalaemonTokens() throws Exception {
                 );
     }
 
+    @Test
+    void shouldGetPreComputePalaemonTokensWithoutDataset() throws Exception {
```

Comment:
Since the session contains critical information, I suggest we assert its exact content not only `contains...`. This is to make sure we add only exactly what needed in the session and it would help detect any non voluntary changes.

Reply:
I'm not sure what you mean there. Do you want us to check that it contains only those 4 keys and their content, but no other key?

Reply:
Yes that's what I mean.

Reply:
Right, see https://github.com/iExecBlockchainComputing/iexec-sms/pull/97/commits/b78f003062d84329973e0c28fb36472e9ba6102f :)